Sources/Sentry/SentrySDK.m

```diff
@@ -135,27 +137,32 @@ + (void)setStartInvocations:(NSUInteger)value
 
 + (void)startWithOptions:(SentryOptions *)options
 {
-    startInvocations++;
+    // We have to call the init on the main thread synchronously to guarantee the SDK gets
+    // initialized. Otherwise, users would init the SDK, and calls directly after would have
+    // undefined behavior as the SDK might not be initialized yet.
+    [SentryThreadWrapper onMainThreadSync:^{
```

Comment:
@philipphofmann, the first idea was to make it sync. But @armcknight and me agreed that making it sync may lead to deadlocks as we had in #3277.

We are aware of undefined behaviour if the user chooses to start it in a background thread, but that would be a user choice. I believe, the best thing we can do here is document this very well.

Reply:
I was also afraid that this might cause deadlocks. We should be fine with the tradeoff if someone inits the SDK on a background thread that it isn't ready immediately before running into a deadlock. I will add some docs.